portfolio/src/main/webapp/book-detail/book-detail.component.js

```diff
@@ -2,10 +2,11 @@ angular.
   module('bookDetail').
   component('bookDetail', {
     templateUrl: 'book-detail/book-detail.template.html',
-    controller: ['$routeParams',
-      function BookDetailController($routeParams) {
-          // This gets the ID from the URL. For example ...#!/book-detail/2  ->  2
-          this.bookId = $routeParams.bookId;
-      }
-    ]
+    controller: function BookDetailController($scope,$http,$routeParams) {
+        this.bookId = $routeParams.bookId;
+      $http.get('book-data')
```

Comment:
Let's change this up!

What I see you're currently doing here:
- You call the book-data servlet to get the list of all books (The same one we call in the book-list component)
- Then, in your view (book-detail.template.html) you use the ID from the parameter to get the exact book you want from the array.

While this works fine, I think we can do better. 1) If the list of all books where really big, we'd spend a lot of time / network resources loading all from the server -- all when we really only want one book. 2) Your view's code becomes more complex than it needs to be since it has to worry about array access.

A cleaner approach:
- Add a new servlet that has a method for getting just one book. You would pass the book ID as a parameter in the request to the servlet.
- Only the book we care about would be returned in the request. (You've already made progress towards this in your servlet by putting all the books in a HashMap! Now just add a method that gets one of them out of the map and returns that one instead of the entire map.)
- Then our view (html) would be much simpler too, since we have just one book instead of a whole array of them.